apps/api/src/modules/webhook/webhook.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Webhook } from './entities/webhook.entity';
+import { Notification } from '../notifications/entities/notification.entity';
+import axios from 'axios';
+
+@Injectable()
+export class WebhookService {
+  protected readonly logger = new Logger(WebhookService.name);
+
+  constructor(
+    @InjectRepository(Webhook)
+    private readonly webhookRepository: Repository<Webhook>,
+  ) {}
+
+  async registerWebhook(providerId: number, webhookUrl: string): Promise<void> {
+    const webhook = new Webhook();
+    webhook.providerId = providerId;
```

Comment:
For now make sure provider_id is unique.
So add check on Add a check to check of records where status is ACTIVE and provider_id is same.

Reply:
added the check